File: OsmAndWear/src/net/osmand/plus/settings/enums/AngularConstants.java
package net.osmand.plus.settings.enums;

import android.content.Context;

import net.osmand.plus.R;

public enum AngularConstants {
	DEGREES(R.string.shared_string_degrees, "°"),
	DEGREES360(R.string.shared_string_degrees, "°"),
	MILLIRADS(R.string.shared_string_milliradians, "mil");

	private final int key;
	private final String unit;

	AngularConstants(int key, String unit) {
		this.key = key;
		this.unit = unit;
	}

	public String toHumanString(Context ctx) {
		return ctx.getString(key);
	}

	public String getUnitSymbol() {
		return unit;
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/settings/enums/AngularConstants.kt
package net.osmand.shared.settings.enums

import net.osmand.shared.units.AngleUnits

enum class AngularConstants {

	DEGREES, DEGREES360, MILLIRADS;

	fun toHumanString() = toUnit().getName()

	fun getUnitSymbol() = toUnit().getSymbol()

	fun toUnit(): AngleUnits {
		return when(this) {
			DEGREES, DEGREES360 -> AngleUnits.DEGREES
			MILLIRADS -> AngleUnits.MILLIRADIANS
		}
	}
}